include/multipass/platform.h

```diff
@@ -58,7 +58,11 @@ class Platform : public Singleton<Platform>
     virtual bool is_remote_supported(const std::string& remote) const;
     virtual bool is_backend_supported(const QString& backend) const; // temporary (?)
     virtual int chown(const char* path, unsigned int uid, unsigned int gid) const;
-    virtual bool set_permissions(const std::filesystem::path& path, std::filesystem::perms permissions) const;
+    virtual bool set_permissions(const std::filesystem::path& path,
+                                 std::filesystem::perms permissions,
+                                 bool try_inherit = false) const;
+    virtual bool take_ownership(const std::filesystem::path& path) const;
```

Comment:
I was looking into our current code for this because of a support question and it occurred to me that a `nodiscard` attribute would help ensure we handle errors appropriately.